src/components/QrcodeScannerDialog.vue

```diff
@@ -75,92 +75,129 @@
 
-export default {
+import { Scanner } from '@/lib/zxing/index'
+
+const className = 'qrcode-scanner-dialog'
+const classes = {
+  root: className,
+  camera: `${className}__camera`,
+  cameraSelect: `${className}__camera-select`
+}
+
+export default defineComponent({
   props: {
     modelValue: {
       type: Boolean,
       required: true
     }
   },
   emits: ['scan', 'update:modelValue'],
-  data: () => ({
-    cameraStatus: 'waiting', // can be: waiting, active, nocamera
-    scanner: null,
-    currentCamera: null,
-    cameras: []
-  }),
-  computed: {
-    className: () => 'qrcode-scanner-dialog',
-    show: {
+  setup(props, { emit }) {
+    const store = useStore()
+    const { t } = useI18n()
+
+    const videoElement = ref<HTMLVideoElement | null>(null)
+    const cameraStatus = ref('waiting') // can be: waiting, active, nocamera
+    const scanner = ref<Scanner | null>(null)
+    const cameras = ref<MediaDeviceInfo[]>([])
+    const currentCamera = ref<string | null>(null)
+    const scannerControls = ref<IScannerControls | null>(null)
+
+    const show = computed<boolean>({
       get() {
-        return this.modelValue
+        return props.modelValue
       },
       set(value) {
-        this.$emit('update:modelValue', value)
+        emit('update:modelValue', value)
       }
-    }
-  },
-  watch: {
-    cameras(cameras) {
-      if (cameras.length > 0) {
-        const cameraKey = cameras.length === 2 ? 1 : 0
-        this.currentCamera = this.cameras[cameraKey].deviceId
+    })
 
-        this.cameraStatus = 'active'
-      } else {
-        this.cameraStatus = 'nocamera'
-      }
-    },
-    currentCamera() {
-      this.scanner.start(this.currentCamera).then((content) => this.onScan(content))
-    }
-  },
-  mounted() {
-    this.init()
-  },
-  beforeUnmount() {
-    this.destroyScanner()
-  },
-  methods: {
-    init() {
-      return this.initScanner()
+    const init = async () => {
+      return initScanner()
         .then(() => {
-          return this.getCameras()
+          return getCameras()
         })
         .catch((err) => {
-          this.cameraStatus = 'nocamera'
-          this.$store.dispatch('snackbar/show', {
-            message: this.$t('scan.something_wrong')
+          cameraStatus.value = 'nocamera'
+          store.dispatch('snackbar/show', {
+            message: t('scan.something_wrong')
           })
           console.error(err)
         })
-    },
-    async initScanner() {
-      this.scanner = new Scanner({
-        videoElement: this.$refs.camera
+    }
+
+    const initScanner = async () => {
+      scanner.value = new Scanner({
+        videoElement: videoElement.value as HTMLVideoElement
       })
 
-      return this.scanner.init()
-    },
-    destroyScanner() {
-      // First check if the scanner was initialized.
+      return scanner.value.init()
+    }
+
+    const getCameras = async () => {
+      cameras.value = await (scanner.value as Scanner).getCameras()
```

Comment:
```suggestion
      cameras.value = await scanner.value!.getCameras()
```

Use `!` instead of type casting